src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,34 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.data.exception.DukeException;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+/**
+ * Finds specific tasks.
+ */
+public class FindCommand extends Command {
+    private String findString;
+
+    /**
+     * Constructor for find command.
+     *
+     * @param findString String to find in tasks.
+     */
+    public FindCommand(String findString) {
+        this.findString = findString;
+    }
+    /**
+     * Executes the command given by the user.
+     *
+     * @param tasks   to be modified
+     * @param ui      to display changes
+     * @param storage to interact with as necessary
+     * @throws DukeException
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        ui.showMatchingTasks(tasks.filterTasks(this.findString));
```

Comment:
I like how you separated the ui and the tasklist such that there's a command for the ui to print each respective action that the tasklist executed, instead of letting the tasklist print it. This embodies the concept of abstraction. 👍

Reply:
Thanks!

> I like how you separated the ui and the tasklist such that there's a command for the ui to print each respective action that the tasklist executed, instead of letting the tasklist print it. This embodies the concept of abstraction. 👍